internal/services/storage/storage_share_resource.go

```diff
@@ -143,239 +150,424 @@ func resourceStorageShare() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if !features.FivePointOhBeta() {
+		r.Schema["storage_account_name"] = &pluginsdk.Schema{
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ForceNew: true,
+			ExactlyOneOf: []string{
+				"storage_account_name",
+				"storage_account_id",
+			},
+			Deprecated: "This property has been deprecated and will be replaced by 'storage_account_id' in version 5.0 of the provider.",
```

Comment:
I think all other deprecation messages begin with lower case, can we be consistent?